src/api_recievers/influxqueries.js

```diff
@@ -308,4 +308,50 @@ const grabAllPreviousStudySessionIDs = () => {
 };
 
 
-module.exports = { appData, SpecificStudySessionProcessing, grabTimesForApp, insertStudySessionData, grabAllPreviousStudySessionIDs, connectToInflux };
\ No newline at end of file
+const deleteStudySession = async (sessionId) => {
+  const deleteAPI = new DeleteAPI(influxClient);
+
+  const start = '1970-01-01T00:00:00Z'; // Start of the range (earliest possible time)
+  const stop = new Date().toISOString(); // End of the range (now)
+  const deleteKey = `_measurement="AppChange" AND QuerySession == ${sessionId}`;
+  let prevSessionDeleted = null; // Initialize outside the loop
+  
+  const headers = {
+    'Authorization': process.env.INFLUXDB_TOKEN,
+    'Content-type': 'application/json'
+  };
+
+  try {
+    await Promise.all(
+      Object.values(influxBuckets).map(async (bucket) => {
+        await deleteAPI.postDelete({
+          org,
+          bucket,
+          body: {
+            start,
+            stop,
+            deleteKey
+          },
+          headers
+        });
+
+        prevSessionDeleted = sessionId;
+
+        console.log(`You have just deleted the session associated with the ID ${sessionId}`);
+
+      })
+    );
+  } catch (error) {
+    if (prevSessionDeleted !== sessionId) {
+      console.error(`Failed to delete session with ID ${sessionId}. Reason: ${error.message}`);
+      throw error; 
+    } else {
+      console.warn(`The session with ID ${sessionId} was already deleted.`);
```

Comment:
We can delete this if statement if we just redirect to a different timeline after deleting (I added code for doing that below), as then it shouldn't be possible to delete the same ID twice.